src/main/io/serial_cli.c

```diff
@@ -2388,20 +2396,22 @@ static void cliSet(char *cmdline)
                 int_float_value_t tmp;
                 switch (valueTable[i].type & VALUE_MODE_MASK) {
                     case MODE_DIRECT: {
-                            int32_t value = 0;
-                            float valuef = 0;
+                            if(strlen(eqptr) > 0 && strspn(eqptr, "0123456789.+-") == strlen(eqptr)) {
```

Comment:
`strlen(eqptr)` could be replaced with `(*eqptr != 0)` 
Strings are null-terminated so if first character is not null we obviously have non-zero length.